internal/services/authorization/parse/role_assignment.go

```diff
@@ -111,6 +111,14 @@ func ConstructRoleAssignmentId(azureResourceId, tenantId string) string {
 	return fmt.Sprintf("%s|%s", azureResourceId, tenantId)
 }
 
+func DestructRoleAssignmentId(id string) (string, string) {
+	parts := strings.Split(id, "|")
+	if len(parts) == 2 {
+		return parts[0], parts[1]
+	}
+	return id, ""
+}
```

Comment:
After internal discussion, this is fine for now. But we will want to create a composite ID type in the `commonids` package in the `go-azure-helpers` repo that can handle cases like these and will replace this in time. Just as an FYI.